robot-core/src/main/java/org/obolibrary/robot/template/TemplateHelper.java

```diff
@@ -153,147 +170,168 @@ public static OWLAnnotation getAnnotation(
-      throws Exception {
-    Set<OWLAxiom> axioms = new HashSet<>();
-    switch (classType) {
-      case "subclass":
-        for (OWLClassExpression expression : classExpressions) {
-          axioms.add(dataFactory.getOWLSubClassOfAxiom(cls, expression));
-        }
-        for (Entry<OWLClassExpression, Set<OWLAnnotation>> annotatedEx :
-            annotatedExpressions.entrySet()) {
-          axioms.add(
-              dataFactory.getOWLSubClassOfAxiom(cls, annotatedEx.getKey(), annotatedEx.getValue()));
-        }
-        return axioms;
-      case "equivalent":
-        // Since it's an intersection, all annotations will be added to the same axiom
-        Set<OWLAnnotation> annotations = new HashSet<>();
-        for (Entry<OWLClassExpression, Set<OWLAnnotation>> annotatedEx :
-            annotatedExpressions.entrySet()) {
-          classExpressions.add(annotatedEx.getKey());
-          annotations.addAll(annotatedEx.getValue());
-        }
-        OWLObjectIntersectionOf intersection =
-            dataFactory.getOWLObjectIntersectionOf(classExpressions);
-        OWLAxiom axiom;
-        if (!annotations.isEmpty()) {
-          axiom = dataFactory.getOWLEquivalentClassesAxiom(cls, intersection, annotations);
-        } else {
-          axiom = dataFactory.getOWLEquivalentClassesAxiom(cls, intersection);
-        }
-        return Sets.newHashSet(axiom);
-      default:
-        throw new Exception(String.format(classTypeError, classType));
+  public static Set<OWLDataPropertyExpression> getDataPropertyExpressions(
+      QuotedEntityChecker checker, String template, String value) {
+    String split = null;
+    if (template.contains("SPLIT=")) {
+      split = template.substring(template.indexOf("SPLIT=") + 6).trim();
+      template = template.substring(0, template.indexOf("SPLIT="));
     }
+
+    // Create a parser
+    ManchesterOWLSyntaxParser parser =
+        new ManchesterOWLSyntaxParserImpl(
+            OWLOntologyLoaderConfiguration::new, OWLManager.getOWLDataFactory());
+    parser.setOWLEntityChecker(checker);
+
+    // Maybe split values
+    List<String> allValues = getAllValues(value, split);
+
+    Set<OWLDataPropertyExpression> expressions = new HashSet<>();
+    if (template.startsWith("PI")) {
+      for (String v : allValues) {
+        String content = QuotedEntityChecker.wrap(v);
+        OWLDataProperty property = checker.getOWLDataProperty(content);
+        expressions.add(property);
+      }
+    } else if (template.startsWith("P ")) {
+      for (String v : allValues) {
+        String content = QuotedEntityChecker.wrap(value);
+        String sub = template.substring(2).trim().replaceAll("%", content);
+        parser.setStringToParse(sub);
+        expressions.addAll(parser.parseDataPropertyList());
+      }
+    }
+
+    return expressions;
   }
 
   /**
-   * Find an annotation property with the given name or create one.
+   * TODO
```

Comment:
TODO what?

Reply:
I was trying to add support for data ranges and it was pretty messy. The current template implementation doesn't support data ranges, but it could be added in the future. I'm going to remove this method for now since it doesn't actually do anything.

Reply:
Ok